crates/crane/src/typer/ty_context.rs

```diff
@@ -2,17 +2,26 @@ use typed_arena::Arena as TypedArena;
 
 use crate::interned::Interned;
 use crate::typer::ty::{Ty, TyKind};
-use crate::typer::UintTy;
+use crate::typer::{FnSig, UintTy};
 
 #[derive(Default)]
 pub struct Arena<'ctx> {
     pub tys: TypedArena<TyKind<'ctx>>,
+    pub ty_lists: TypedArena<Vec<Ty<'ctx>>>,
 }
 
 impl<'ctx> Arena<'ctx> {
     pub fn intern_ty(&self, kind: TyKind<'ctx>) -> Ty {
         Ty(Interned::new_unchecked(self.tys.alloc(kind)))
     }
+
+    pub fn intern_fn(&'ctx self, params: Vec<Ty<'ctx>>, return_ty: Ty<'ctx>) -> Ty<'ctx> {
+        let params = self.ty_lists.alloc(params);
```

Comment:
Since we need to store a list of parameters for the function signature, we intern the parameter list in a separate arena and then store the reference to that interned list in the `FnSig` itself.

Both arenas have the same lifetime, so the values will all get dropped at the same time.